bundler/spec/commands/fund_spec.rb

```diff
@@ -55,6 +55,41 @@
     expect(out).to_not include("gem_with_dependent_funding")
   end
 
+  it "does not consider fund information for uninstalled optional dependencies" do
+    install_gemfile <<-G
+      source "#{file_uri_for(gem_repo2)}"
+      group :whatever, optional: true do
+        gem 'has_funding_and_other_metadata'
+      end
+      gem 'has_funding'
+      gem 'rack-obama'
+    G
+
+    bundle "fund"
+
+    expect(out).to include("* has_funding (1.2.3)\n  Funding: https://example.com/has_funding/funding")
+    expect(out).to_not include("has_funding_and_other_metadata")
+    expect(out).to_not include("rack-obama")
+  end
+
+  it "considers fund information for installed optional dependencies" do
+    install_gemfile <<-G
+      source "#{file_uri_for(gem_repo2)}"
+      group :whatever, optional: true do
+        gem 'has_funding_and_other_metadata'
+      end
+      gem 'has_funding'
+      gem 'rack-obama'
+    G
+
+    bundle "install", with: "whatever"
```

Comment:
Is `with: "whatever"` needed? The `--with` flag will be removed in Bundler 3, so Bundler 3 specs raise an error here.

Reply:
Ah, right. Yeah, those are going away. It's needed since only installed gems are considered for funding. Switched it over to using the env variable instead.